pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -246,12 +246,12 @@ public void run() {
 
     private void loadJars() throws Exception {
 
-        if (jarFile.endsWith(".nar")) {
-            // The functions code is contained in a NAR archive
+        try {
+            // Let's first try to treat it as a nar archive
             fnCache.registerFunctionInstanceWithArchive(instanceConfig.getFunctionId(), instanceConfig.getInstanceId(),
                     jarFile);
-        } else {
-            log.info("Loading JAR files for function {} from archive {}", instanceConfig, jarFile);
+        } catch (Exception e) {
+            log.info("For Function {} Loading as NAR failed with {}; treating it as Jar instead", instanceConfig, e);
```

Comment:
can we verify if it actually throws exception with normal jar file as well?

Reply:
sources/sinks can be submitted as either .nar or .jar. And this will work in both cases.
Maybe I don't understand your comment?

Reply:
i mean now, this change identifies it as a jar if `fnCache.registerFunctionInstanceWithArchive` throws exception. so, just wanted to check if we can test with real jar file and see `fnCache.registerFunctionInstanceWithArchive`  is actually throwing exception or eating up internally.